File: backend/DormitoryManagement.Data/Models/Mattress.cs

using System.ComponentModel.DataAnnotations;

namespace DormitoryManagement.Data.Models
{
    public partial class Mattress
    {
        [Key]
        [MaxLength(50)]
        public int SerialNumber { get; set; }
        [MaxLength(50)]
        public int ConditionId { get; set; }
        public Condition Condition { get; set; }
        [MaxLength(50)]
        public int TypeId { get; set; }
        public MattressType Type { get; set; }
        [MaxLength(50)]
        public string StudentNumber { get; set; }
        public Student? Student { get; set; }
    }
}

File: frontend/src/app/models/Mattress.ts
import {Condition, MattressType} from './Enums';

export interface Mattress {
  serialNumber: number;
  condition: Condition;
  type: MattressType;
  studentNumber: string;
  expanded?: boolean;
}

export interface MattressDTO {
  serialNumber?: number; // Опціонально, бо не завжди потрібен при оновленні
  conditionId: number;
  typeId: number;
  studentNumber: string;
}
